src/Microsoft.ML.TimeSeries/IidChangePointDetector.cs

```diff
@@ -72,23 +71,23 @@ public BaseArguments(Arguments args)
             {
                 Source = args.Source;
                 Name = args.Name;
-                Side = SequentialAnomalyDetectionTransformBase<float, State>.AnomalySide.TwoSided;
+                Side = AnomalySide.TwoSided;
                 WindowSize = args.ChangeHistoryLength;
                 Martingale = args.Martingale;
                 PowerMartingaleEpsilon = args.PowerMartingaleEpsilon;
-                AlertOn = SequentialAnomalyDetectionTransformBase<float, State>.AlertingScore.MartingaleScore;
+                AlertOn = AlertingScore.MartingaleScore;
             }
 
             public BaseArguments(IidChangePointDetector transform)
```

Comment:
This is a begining of circular dependency. Looks like this `BaseArguments` is used to create `IidChangePointDetector` and `IidChangePointDetector` can be used to construct `BaseArgument`. Can we have a constructor which is able to set all fields in this class as an alternative solution?

Reply:
I have made all this internal now.

---
In reply to: [254486695](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254486695) [](ancestors = 254486695)

Reply:
Making this internal doesn't mean that ML.NET team is not able to create circular dependency. Could you please fix it?

---
In reply to: [255177913](https://github.com/dotnet/machinelearning/pull/2344#discussion_r255177913) [](ancestors = 255177913,254486695)